### DISCUSSION TITLE
log-rotate question

### DISCUSSION THREAD

**Discussion by dimon-v** [apache/apisix#5361] [Q&A]
Apisix version: 2.9.0/2.10.0
Etcd version: 3.5.0
log-ratate url: https://apisix.apache.org/docs/apisix/plugins/log-rotate/

I needed to add a specific identifier to the HTTP response header. 
When I added 'plugins: - log-rotate' to the config.yaml file, following the instructions in the official documentation,
After log splitting is enabled, plug-ins that take effect previously, such as Response-rewrite and kafka-logger, become invalid  
 
Do you need to add all used plug-ins to config.yaml?

**Comment by tokers** [comment]
> Apisix version: 2.9.0/2.10.0 Etcd version: 3.5.0 log-ratate url: https://apisix.apache.org/docs/apisix/plugins/log-rotate/
> 
> I needed to add a specific identifier to the HTTP response header. When I added 'plugins: - log-rotate' to the config.yaml file, following the instructions in the official documentation, After log splitting is enabled, plug-ins that take effect previously, such as Response-rewrite and kafka-logger, become invalid
> 
> Do you need to add all used plug-ins to config.yaml?

Did you remove other plugins from the plugin list?

**Reply by dimon-v** [reply]
No modification config-default.yaml, just adding config.yaml (plugins: - log-rotate)

-- config.yaml

plugins:
    - log-rotate

plugin_attr:
    log-rotate:
        interval: 3600    # rotate interval (unit: second)
        max_kept: 168     # max number of log files will be kept
        enable_compression: false    # enable log file compression(gzip) or not, default false
